cpp/include/cudf/detail/aggregation/aggregation.hpp

```diff
@@ -866,10 +873,10 @@ class lead_lag_aggregation final : public rolling_aggregation {
 class udf_aggregation final : public rolling_aggregation {
  public:
   udf_aggregation(aggregation::Kind type,
-                  std::string const& user_defined_aggregator,
+                  std::string user_defined_aggregator,
```

Comment:
Should this take a `string_view` instead of a `string` that we move? This happens quite a few times in this PR. I understand the potential (small?) benefit of move-semantics here, but it feels more idiomatic to accept a `string_view`.

Reply:
I agree, but I'm not sure I want to attempt that in this PR. Would you be OK with opening an issue to replace `std::string` parameters with `std::string_view` and doing it in a follow-up?

Reply:
That sounds fine to me. Totally happy to defer on this.

Reply:
https://github.com/rapidsai/cudf/issues/15907